tests/cross_functional/system_test/test_mcg_replication_with_disruptions.py

```diff
@@ -228,3 +241,150 @@ def test_replication_with_disruptions(
         logger.info("Objects sync works even when the cluster is rebooted")
+
+
+@system_test
+@magenta_squad
+@skipif_vsphere_ipi
+class TestLogBasedReplicationWithDisruptions:
+    @polarion_id("OCS-5457")
+    @bugzilla("2266805")
+    def test_log_based_replication_with_disruptions(
+        self,
+        mcg_obj_session,
+        aws_log_based_replication_setup,
+        noobaa_db_backup,
+        noobaa_db_recovery_from_backup,
+        setup_mcg_bg_features,
+        validate_mcg_bg_features,
+    ):
+        """
+        This is a system test flow to test log based bucket replication
+        deletion sync is not impacted due to some noobaa specific disruptions
+        like noobaa pod restarts, noobaa db backup & recovery etc
+
+        1. Setup log based bucket replication between the buckets
+        2. Upload some objects and make sure replication works
+        3. Keep deleting some objects from the source bucket and make sure
+           deletion sync works as expected through out.
+        4. In another thread, restart the noobaa pods (db & core), make sure
+           deletion sync works for the step-3 deletion works as expected
+        5. Now take backup of Noobaa db using PV backup method
+        6. Remove the log based replication rules, perform some deletion in
+           source bucket. make sure deletion sync doesn't work
+        7. Recover noobaa db from the backup taken in step-5
+        8. Now check if deletion sync works by deleting some objects from
+           source bucket. Note: Expectation is still unclear
+        9. Now patch the bucket to remove complete replication policy and
+           make sure no replication - no deletion sync works
+
+        """
+
+        # entry criteria setup
+        feature_setup_map = setup_mcg_bg_features(
+            num_of_buckets=5,
+            object_amount=5,
+            is_disruptive=True,
+            skip_any_features=["nsfs", "rgw kafka", "caching", "replication"],
+        )
+
+        mockup_logger, source_bucket, target_bucket = aws_log_based_replication_setup()
+
+        # upload test objects to the bucket and verify replication
+        upload_test_objects_to_source_and_wait_for_replication(
+            mcg_obj_session,
+            source_bucket,
+            target_bucket,
+            mockup_logger,
+            600,
+        )
+
+        # Delete objects in the first set in a batch and perform noobaa pod
+        # restarts at the same time and make sure deletion sync works
+
+        objs_in_bucket = mockup_logger.standard_test_obj_list
+        objs_to_delete = random.sample(objs_in_bucket, 3)
+
+        from ocs_ci.utility.retry import retry
+
+        @retry(Exception, tries=10, delay=5)
+        def delete_objs_in_batch():
+            for obj in objs_to_delete:
+                mockup_logger.delete_objs_and_log(source_bucket.name, [obj])
+                time.sleep(5)
+            logger.info(f"Successfully deleted these objects: {objs_to_delete}")
```

Comment:
It is not a good practice to define the functions inside a test function. Can you move it out